AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginAuthIntegrationTests/MultiAuth/AWSDataStoreMultiAuthSingleRuleTests.swift

```diff
@@ -0,0 +1,191 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import Foundation
+import Combine
+
+@testable import Amplify
+@testable import AmplifyPlugins
+@testable import AmplifyTestCommon
+
+class AWSDataStoreMultiAuthSingleRuleTests: AWSDataStoreMultiAuthBaseTest {
+    /// Given: a user signed in with CognitoUserPools
+    /// When: DataStore query/mutation operations are sent with CognitoUserPools
+    /// Then: DataStore is successfully initialized, query returns a result,
+    ///      mutation is processed for an authenticated users
+    func testOwnerUserPools() {
+        setup(withModels: UserPoolsOwnerModels())
+
+        signIn(user: user1)
+
+        let expectations = makeExpectations()
+
+        assertDataStoreReady(expectations)
+
+        // Query
+        assertQuerySuccess(modelType: OwnerUPPost.self,
+                           expectations) { error in
+            XCTFail("Error query \(error)")
+        }
+
+        // Mutations
+        assertMutations(model: OwnerUPPost(name: "name"), expectations) { error in
+            XCTFail("Error mutation \(error)")
+        }
+
+        assertUsedAuthTypes([.amazonCognitoUserPools])
+    }
+
+    /// Given: a user signed in with OIDC
+    /// When: DataStore query/mutation operations are sent with OIDC
+    /// Then: DataStore is successfully initialized, query returns a result,
+    ///      mutation is processed for an authenticated users
+    func testOwnerOIDC() throws {
+        // PLACEHOLDER
+        throw XCTSkip("Not implemented")
+    }
+
+    /// Given: a user signed in with CognitoUserPools
+    /// When: DataStore query/mutation operations are sent with CognitoUserPools
+    /// Then: DataStore is successfully initialized, query returns a result,
+    ///      mutation is processed for an authenticated users
+    func testGroupUserPools() {
```

Comment:
is there a good place to put a negative scenario test? sign in with user2, they are not in the group, DataStore starts and fails to sync that data, however datastore starts successfully. They try to save the GroupUPPost, it saves to local store, and fails to save to remote store because they are not in the group. errorHandler gets called and error is that they are unauthorized. 

what happens to local data btw?

Reply:
Updated with an additional test case with `user2`